File: API/Services/BudgetService.cs
using System.Collections.Generic;
using API.Models;
using API.Repositories;

namespace API.Services
{
    public class BudgetService
    {
        private readonly BudgetRepository _repo;

        public BudgetService(BudgetRepository repo)
        {
            _repo = repo;
        }

        public Budget CreateBudget(Budget newBudget)
        {
            return _repo.CreateBuget(newBudget);
        }

        public List<Budget> GetBudgets(int userId)
        {
            return _repo.GetUserBudgets(userId);
        }
    }
}

File: client/src/app/budget/budget.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Budget } from "../shared/models/budget.model";
import { BudgetDTO } from "./budget-form/budget-form.component";

@Injectable({
  providedIn: 'root'
})
export class BudgetService {
  budgets = [];

  constructor(private http: HttpClient) {}

  createBudget(budget: BudgetDTO) {
    this.http.post<Budget>('http://localhost:5000/api/budgets', budget)
      .subscribe({
        next: res => {
          console.log(res)
        },
        error: err => console.log(err)
      })
  }

  getUserBudgets(userId: number) {
    this.http.get<Budget[]>(`http://localhost:5000/api/budgets/user/${userId}`)
      .subscribe({
        next: res => {
          console.log(res)
        },
        error: err => console.log(err)
      })
  }
}
